test/reporter-html/diff.js

```diff
@@ -0,0 +1,163 @@
+QUnit.module( "diff" );
+
+QUnit.test( "throws if arguments are not strings", function( assert ) {
+	assert.throws(function() { QUnit.diff( {}, "" ); });
+	assert.throws(function() { QUnit.diff( "", {} ); });
+
+	assert.throws(function() { QUnit.diff( [], "" ); });
+	assert.throws(function() { QUnit.diff( "", [] ); });
+
+	assert.throws(function() { QUnit.diff( undefined, "" ); });
+	assert.throws(function() { QUnit.diff( "", undefined ); });
+
+	assert.throws(function() { QUnit.diff( null, "" ); });
+	assert.throws(function() { QUnit.diff( "", null ); });
+
+	assert.throws(function() { QUnit.diff( 42, "" ); });
+	assert.throws(function() { QUnit.diff( "", 42 ); });
+
+	assert.throws(function() { QUnit.diff( true, "" ); });
+	assert.throws(function() { QUnit.diff( "", true ); });
+});
+
+QUnit.test( "different strings", function( assert ) {
+	var a = "abcd";
+	var b = "xkcd";
+
+	assert.equal(
+		QUnit.diff( a, b ),
+		"<del>ab</del><ins>xk</ins><span>cd</span>",
+		"QUnit.diff( a, b )"
```

Comment:
This doesn't seem like a particularly useful message. Also applies to the others.